config/fi_provider.m4

```diff
@@ -125,7 +125,8 @@ AC_DEFUN([FI_PROVIDER_SETUP],[
 	AS_IF([test x"$enable_direct" = x"$1"],
 		[AC_MSG_CHECKING(for $srcdir/prov/$1/include/rdma/fi_direct.h)
 		 AS_IF([test -f "$srcdir/prov/$1/include/rdma/fi_direct.h"],
-			[AC_MSG_RESULT(yes)],
+			[AC_MSG_RESULT(yes)
+			CFLAGS="$CFLAGS -DFI_DIRECT_ENABLED"],
 			[AC_MSG_RESULT(no)
```

Comment:
I think it would be better to use an AC_DEFINE_UNQUOTED here or in the top level configure.ac rather than setting CFLAGS.